### PR TITLE
feat: add support of mongodb v3.7.x or higher

### PR DESCRIPTION
**Pull request by scolia**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

add support of mongodb v3.7.x or higher.

### CODE REVIEW

**Comment by wu-sheng** [comment]
We will expect you to move lower versions tests with these versions tests in this PR.

Take refs in 
- #3575, Compose test cases are supported soon
- #3584 
- Existing test cases on in https://github.com/SkyAPMTest/agent-auto-integration-testcases/tree/master/mongodb-3.x-scenario. Move this test into the PR and add your new versions.

Ping me on QQ or WeChat, this test is still experimental, so no full document, you may need helps directly.

**Comment by dmsolr** [comment]
refer to #3583

**Comment by YunaiV** [comment]
🐂🍺just need

Whether intercept CommandOperationHelper can be considered?
params `connection`、`database`、`command` is easy to use 。

**Comment by wu-sheng** [comment]
Related to #2649

**Comment by wu-sheng** [comment]
> Move this test into the PR and add your new versions.

The new suggestion is, move the old existing tests in a separated PR, and after it is merged, we continue to work on this one.

**Comment by wu-sheng** [comment]
Existing mongodb plugin could be moved into, now

File: apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/define/v30/MongoDBV30Instrumentation.java [hunk 1]
```diff
@@ -17,67 +17,86 @@
...
 
 import net.bytebuddy.description.method.MethodDescription;
 import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
 import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
 import org.apache.skywalking.apm.agent.core.plugin.match.NameMatch;
-import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.plugin.mongodb.v3.interceptor.v30.MongoDBV30Interceptor;
```
Comment (dmsolr) [inline apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/define/v30/MongoDBV30Instrumentation.java]: I took a glance. I think this rename class as `org.apache.skywalking.apm.plugin.mongodb.v30.interceptor.MongoDBInterceptor` will be more clear. HDYT?

File: test/plugin/scenarios/mongodb-3.x-scenario/support-version.list [hunk 2]
```diff
@@ -19,3 +19,15 @@
 3.4.2
 3.4.3
 3.5.0
+3.7.0
```
Comment (wu-sheng) [inline test/plugin/scenarios/mongodb-3.x-scenario/support-version.list]: Just a context, what happens in 3.5.0-3.7.0? Is the code different? Like an anonymous class used?

File: Jenkinsfile-Agent-Test-3 [hunk 3]
```diff
@@ -96,7 +96,7 @@ pipeline {
                                 sh 'bash test/plugin/run.sh --build_id=wl3_${BUILD_ID} sofarpc-scenario'
                             }
                         }
-                        stage('mongodb 3.0-3.5.0 (5)') {
+                        stage('mongodb 3.0-3.11.1 (17)') {
```
Comment (wu-sheng) [inline Jenkinsfile-Agent-Test-3]: The stage name is not accurate. You are not testing all version in 3.0-3.11. Please change to match the supported-list doc and your real test cases.

File: test/plugin/scenarios/mongodb-3.x-scenario/support-version.list (hunk 2 above)
Comment (scolia) [reply test/plugin/scenarios/mongodb-3.x-scenario/support-version.list]: 3.6.x using anonymous class to implements OperationExecutor, it is difficult to intercept it.

File: Jenkinsfile-Agent-Test-3 (hunk 3 above)
Comment (scolia) [reply Jenkinsfile-Agent-Test-3]: fix

File: test/plugin/scenarios/mongodb-3.x-scenario/support-version.list (hunk 2 above)
Comment (wu-sheng) [reply test/plugin/scenarios/mongodb-3.x-scenario/support-version.list]: Good to doc here. I just want official confirmation

Comment (wu-sheng) [reply test/plugin/scenarios/mongodb-3.x-scenario/support-version.list]: Not in source code, just in the PR should be good. Don't need to update.

**Pull request merged**